sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -224,13 +224,6 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
         table,
         Some(t.location),
         actualQuery.output.map(_.name))
-
-      // For dynamic partition overwrite, we do not delete partition directories ahead.
-      // We write to staging directories and move to final partition directories after writing
-      // job is done. So it is ok to have outputPath try to overwrite inputpath.
-      if (overwrite && !insertCommand.dynamicPartitionOverwrite) {
-        DDLUtils.verifyNotReadPath(actualQuery, outputPath)
```

Comment:
we can remove this method now?

Reply:
> we can remove this method now?

Yes, since this check is not correct and will  block SQL that can be executed properly